pupil_src/shared_modules/gaze_mapping/gazer_2d_v2.py

```diff
@@ -0,0 +1,246 @@
+                    f"Fitting. RMSE = {rmse:>7.2f}px ..."
+                    f" discarding {n_filtered_out}/{X.shape[0]}"
+                    f" ({100 * (n_filtered_out) / X.shape[0]:.2f}%)"
+                    f" data points as outliers."
+                )
+                # recursively remove outliers
+                return self.fit(
+                    X_filtered,
+                    Y_filtered,
+                    outlier_removal_iterations=outlier_removal_iterations - 1,
+                )
+
+        logger.debug(f"Fitting. RMSE = {rmse:>7.2f}px in final iteration.")
+        self._is_fitted = True
+
+    def _test_pixel_error(self, X, Y):
+        Y_predict = self.predict(X)
+        difference_px = (Y_predict - Y) * self.screen_size
+        errors_px = np.linalg.norm(difference_px, axis=1)
+        root_mean_squared_error_px = np.sqrt(np.mean(np.square(errors_px)))
+        return errors_px, root_mean_squared_error_px
+
+    def predict(self, X):
+        self._validate_feature_dimensionality(X)
+        polynomial_features = self._polynomial_features(X)
+        return self._regressor.predict(polynomial_features)
+
+    def _polynomial_features(self, norm_xy):
+        # slice data to retain ndim
+        norm_x = norm_xy[:, :1]
+        norm_y = norm_xy[:, 1:]
+
+        norm_x_squared = norm_x ** 2
+        norm_y_squared = norm_y ** 2
+
+        return np.hstack(
+            (
+                norm_x,
+                norm_y,
+                norm_x * norm_y,
+                norm_x_squared,
+                norm_y_squared,
+                norm_x_squared * norm_y_squared,
+            )
+        )
+
+    @staticmethod
+    def _validate_reference_dimensionality(Y):
+        assert Y.ndim == 2, "Required shape: (n_samples, n_features)"
+        assert Y.shape[1] == _REFERENCE_FEATURE_COUNT
+
+    @staticmethod
+    @abc.abstractmethod
+    def _validate_feature_dimensionality(X):
+        raise NotImplementedError
+
+
+class Model2D_v2_Binocular(Model2D_v2):
+    def _polynomial_features(self, norm_xy):
+        left = super()._polynomial_features(norm_xy[:, _MONOCULAR_PUPIL_NORM_POS])
+        right = super()._polynomial_features(norm_xy[:, _BINOCULAR_PUPIL_NORM_POS])
+        return np.hstack((left, right))
+
+    @staticmethod
+    def _validate_feature_dimensionality(X):
+        assert X.ndim == 2, "Required shape: (n_samples, n_features)"
+        assert X.shape[1] == _BINOCULAR_FEATURE_COUNT, (
+            f"Received shape: {X.shape}. "
+            f"Expected shape (n_samples, {_BINOCULAR_FEATURE_COUNT})"
+        )
+
+
+class Model2D_v2_Monocular(Model2D_v2):
+    @staticmethod
+    def _validate_feature_dimensionality(X):
+        assert X.ndim == 2, "Required shape: (n_samples, n_features)"
+        assert X.shape[1] == _MONOCULAR_FEATURE_COUNT, (
+            f"Received shape: {X.shape}. "
+            f"Expected shape (n_samples, {_MONOCULAR_FEATURE_COUNT})"
+        )
+
+
+class Gazer2D(GazerBase):
```

Comment:
can we rename this to `Gazer2D_v2x` to be consistent with `Gazer2D_v1x`? also, the module should follow the `v1` pattern and be called `gazer_2d_v2x`.

To be clear, I don't prefer the convention of `v1` over any other convention; just want things to be named consistently.

Reply:
@papr what are your thoughts on this? I understood that we will get rid of the v1 anyways? Do we want to keep this versioning?

Reply:
We will remove the `_vYx` suffix before the release. Specifically, we will remove `Gazer2D_v1x` completely, and change `Gazer3D_v1x` to `Gazer3D`. I just need to make another pass through the code before committing to the final version.

Reply:
Ok so I see this as resolved

Reply:
No wait. Currently the filename is `gazer_2d_v2.py`. Do we also want to change that to `gazer_2d.py` @papr ?

Reply:
I just did this. If that's not ok, pls comment.

Reply:
Correct call